resources/monitor.rb

```diff
@@ -1,15 +1,64 @@
 # Configure a service via its yaml file
 
-actions :add, :remove
-default_action :add if defined?(default_action)
+default_action :add
 
-attribute :name, :kind_of => String, :name_attribute => true
-attribute :cookbook, :kind_of => String, :default => 'datadog'
+property :name, String, name_attribute: true
+property :cookbook, String, default: 'datadog'
 
 # checks have 2 sections: init_config and instances
 # we mimic these here, no validation is performed until the template
 # is evaluated.
-attribute :init_config, :kind_of => Hash, :required => false, :default => {}
-attribute :instances, :kind_of => Array, :required => false, :default => []
-attribute :version, :kind_of => Integer, :required => false, :default => nil
-attribute :use_integration_template, :kind_of => [TrueClass, FalseClass], :required => false, :default => false
+property :init_config, Hash, required: false, default: {}
+property :instances, Array, required: false, default: []
+property :version, Integer, required: false, default: nil
+property :use_integration_template, [TrueClass, FalseClass], required: false, default: false
+
+action :add do
+  log "Adding monitoring for #{new_resource.name}" do
+    level :debug
+  end
+
+  template ::File.join(node['datadog']['config_dir'], 'conf.d', "#{new_resource.name}.yaml") do
+    if node['platform_family'] == 'windows'
+      owner 'Administrators'
+      rights :full_control, 'Administrators'
+      inherits false
+    else
+      owner 'dd-agent'
+      mode '600'
+    end
+
+    source 'integration.yaml.erb' if new_resource.use_integration_template
+
+    variables(
+      init_config: new_resource.init_config,
+      instances:   new_resource.instances,
+      version:     new_resource.version
+    )
+    cookbook new_resource.cookbook
+    sensitive true
+    notifies :restart, 'service[datadog-agent]', :delayed if node['datadog']['agent_start']
+  end
+
+  service 'datadog-agent' do
```

Comment:
Since this PR will require a major version bump of the cookbook, I think it's time to remove the `service` resource from the custom resource and make invocations of `datadog_monitor` notify `service[datadog-agent]` directly instead (see #467 and #323 for some background on the current related issue).

Reply:
Ah, that is fair 👍 

Will fix in another commit.